tests/setupconfig/test_auth.py

```diff
@@ -0,0 +1,227 @@
+    ADMIN_OIDC_OIDC_NONCE_SIZE=None,
+    ADMIN_OIDC_OIDC_STATE_SIZE=None,
+    ADMIN_OIDC_OIDC_EXEMPT_URLS=None,
+    ADMIN_OIDC_USERINFO_CLAIMS_SOURCE=None,
+)
+@pytest.mark.django_db
+def test_configure_use_defaults():
+
+    AdminOIDCConfigurationStep().configure()
+
+    config = OpenIDConnectConfig.get_solo()
+
+    assert config.enabled
+    assert config.oidc_rp_client_id == "client-id"
+    assert config.oidc_rp_client_secret == "secret"
+    assert config.oidc_rp_scopes_list == ["openid", "email", "profile"]
+    assert config.oidc_rp_sign_algo == "HS256"
+    assert config.oidc_rp_idp_sign_key == ""
+    assert config.oidc_op_discovery_endpoint == ""
+    assert (
+        config.oidc_op_jwks_endpoint
+        == f"{IDENTITY_PROVIDER}protocol/openid-connect/certs"
+    )
+    assert (
+        config.oidc_op_authorization_endpoint
+        == f"{IDENTITY_PROVIDER}protocol/openid-connect/auth"
+    )
+    assert (
+        config.oidc_op_token_endpoint
+        == f"{IDENTITY_PROVIDER}protocol/openid-connect/token"
+    )
+    assert (
+        config.oidc_op_user_endpoint
+        == f"{IDENTITY_PROVIDER}protocol/openid-connect/userinfo"
+    )
+    assert config.username_claim == ["sub"]
+    assert config.groups_claim == ["groups_claim_name"]
+    assert config.claim_mapping == {
+        "last_name": ["family_name"],
+        "first_name": ["given_name"],
+    }
+    assert config.sync_groups
+    assert config.sync_groups_glob_pattern == "*"
+    assert list(group.name for group in config.default_groups.all()) == [
+        "Admins",
+        "Read-only",
+    ]
+    assert not config.make_users_staff
+    assert config.superuser_group_names == ["superuser"]
+    assert config.oidc_use_nonce
+    assert config.oidc_nonce_size == 32
+    assert config.oidc_state_size == 32
+    assert (
+        config.userinfo_claims_source == UserInformationClaimsSources.userinfo_endpoint
+    )
+
+
+@pytest.fixture
+def discovery_endpoint_response():
+
+    return {
+        "issuer": IDENTITY_PROVIDER,
+        "authorization_endpoint": f"{IDENTITY_PROVIDER}protocol/openid-connect/auth",
+        "token_endpoint": f"{IDENTITY_PROVIDER}protocol/openid-connect/token",
+        "userinfo_endpoint": f"{IDENTITY_PROVIDER}protocol/openid-connect/userinfo",
+        "end_session_endpoint": f"{IDENTITY_PROVIDER}protocol/openid-connect/logout",
+        "jwks_uri": f"{IDENTITY_PROVIDER}protocol/openid-connect/certs",
+    }
+
+
+@override_settings(
+    ADMIN_OIDC_OIDC_OP_DISCOVERY_ENDPOINT=IDENTITY_PROVIDER,
+    ADMIN_OIDC_OIDC_OP_JWKS_ENDPOINT=None,
+    ADMIN_OIDC_OIDC_OP_AUTHORIZATION_ENDPOINT=None,
+    ADMIN_OIDC_OIDC_OP_TOKEN_ENDPOINT=None,
+    ADMIN_OIDC_OIDC_OP_USER_ENDPOINT=None,
+)
+@pytest.mark.django_db
+def test_configure_use_discovery_endpoint(requests_mock, discovery_endpoint_response):
+    requests_mock.get(
+        f"{IDENTITY_PROVIDER}.well-known/openid-configuration",
+        json=discovery_endpoint_response,
+    )
+
```

Comment:
please stop using requests-mock and instead use VCR - there is a working docker-compose setup in this library with keycloak that you can use & a fixture that already makes use of this too: https://github.com/maykinmedia/mozilla-django-oidc-db/blob/d0630df6a5dc9d8bf34dbed5a0be983973779ed6/tests/conftest.py#L70. Example usage: https://github.com/maykinmedia/mozilla-django-oidc-db/blob/d0630df6a5dc9d8bf34dbed5a0be983973779ed6/tests/test_integration_multiple_configs.py#L14